### PR TITLE
DOC: update documentation for the NumFOCUS Sponsored project status

### PR DESCRIPTION
**Pull request by martinfleis**
GeoPandas has been accepted as a NumFOCUS Sponsored project. As part of the onboarding, we were given a checklist of things to update and mention either in the ReadMe or in the documentation. This PR includes them all.

The [guide](https://github.com/numfocus/templates/blob/master/fiscal-sponsor-readme-attribution.md) says that we should have a `GOVERNANCE.md` file in the root but I am linking directly to the document as it lives in its own repository to cover all of the GeoPandas projects, not only the main repo. I suppose it is okay.

The links to geopandas project page and donation page do not work yet but I suppose this is how they'll look like.

### CODE REVIEW

**Comment by martinfleis** [comment]
All the links now correctly lead to respective NumFOCUS pages.

File: README.md [hunk 1]
```diff
@@ -32,6 +33,20 @@ Documentation is available at [geopandas.org](http://geopandas.org)
 [Read the Docs](http://geopandas.readthedocs.io/en/latest/)
 (release and development versions).
 
+The GeoPandas project uses an [open governance model](https://github.com/geopandas/governance/blob/main/Governance.md)
```
Comment (m-richards) [inline README.md:36]:
```suggestion
[//]: # (numfocus-fiscal-sponsor-attribution)

The GeoPandas project uses an [open governance model](https://github.com/geopandas/governance/blob/main/Governance.md)
```

Just missing the markdown comment which is required for automated checks (https://github.com/numfocus/templates/blob/master/fiscal-sponsor-readme-attribution.md?plain=1)

**Review by m-richards — APPROVED** [review]
One small comment, looks good otherwise. I think this has everything required.

**Comment by jorisvandenbossche** [comment]
Thanks!

**Pull request merged**